src/init.c

```diff
@@ -511,12 +518,28 @@ static void hyper_print_uptime(void)
 {
 	char buf[128];
 	int fd = open("/proc/uptime", O_RDONLY);
+	ssize_t bytes_read;
 
 	if (fd < 0)
 		return;
+
 	memset(buf, 0, sizeof(buf));
-	if (read(fd, buf, sizeof(buf)))
+	buf[sizeof(buf)-1] = '\0';
+
+	bytes_read = read(fd, buf, sizeof(buf)-1);
+	if (bytes_read < 0) {
+		fprintf(stdout, "reading /proc/uptime failed: %s\n",
```

Comment:
Best to define a variable for `/proc/uptime` to avoid hard-coding as it's also used above.